lib/filterx/filterx-scope.c

```diff
@@ -356,12 +420,21 @@ _free(FilterXScope *self)
 {
   /* NOTE: update the number of inlined variable allocations */
   gint variables_max = filterx_scope_variables_max;
-  if (variables_max < self->variables->len)
-    filterx_scope_variables_max = self->variables->len;
+  if (variables_max < self->variables.len)
+    filterx_scope_variables_max = self->variables.len;
```

Comment:
I think we should put a hard-coded upper limit to this variable.

If someone loads a configuration with millions of variables, not even a reload will help reduce the memory usage.

Reply:
I am not sure how that could happen, we don't have dynamic allocation of variables, we do not have loops, etc. The only way to create 1M variables is to create a configuration with 1M lines, each defining a variable.

With that said, the usual stack size is 8M, a FilterXVariable is 16 bytes, but we need stack space for the actual function calls, and as we just discussed, FilterXScopes are embedded on the stack. What if we just limited the number of maximum variables any FilterXScope can support to 1024? Or maybe 4096? I don't think anyone would need more, or if they do, we can always bump that higher.

That should limit the stack use to 64kB

Reply:
Maybe I missed something, but I'm only talking about the maximum number of variables that we allocate on the stack. Heap-allocated variables could remain infinite in my opinion.

My suggested would be to rename and calculate `filterx_scope_variables_max` like this:
`filterx_scope_variables_max_inlined = MIN(self->variables.len, 1024)`